### ISSUE TITLE
Content cookie available after delete on Safari

### ISSUE THREAD

**Issue by ohrrkan** [withastro/astro#11779]
### Astro Info

```block
Astro                    v4.14.2
Node                     v20.16.0
System                   macOS (arm64)
Package Manager          npm
Output                   hybrid
Adapter                  @astrojs/vercel/serverless
Integrations             @astrojs/sitemap
                         @astrojs/tailwind
```

### If this issue only occurs in one browser, which browser is a problem?

Safari

### Describe the Bug

Not certain it is a Astro issue or Safari (Need more investigation) but impact. 

The content of the cookie can be access on the client after a cookie.delete (Only on Safari)

Example : 

set.astro
```
---
Astro.cookies.set("sb-access-token", "token", {
  path: "/",
});
---

<!doctype html>
<html>
  <body>
    <h1>Set cookie</h1>
  </body>
</html>

```

delete.astro
```
---
Astro.cookies.delete("sb-access-token", { path: "/" });
---

<!doctype html>
<html>
  <body>
    <h1>Delete cookie</h1>
    <script>
      console.log(document.cookie);
    </script>
  </body>
</html>

```

### What's the expected result?

Tempory solution : `cookies.set("sb-access-token", "",{ path: "/" }); `in place of `  cookies.delete("sb-access-token", { path: "/" });`

### Link to Minimal Reproducible Example

[stackblitz not work with safari](https://stackblitz.com/edit/github-yacene?file=src%2Fpages%2Fremove.astro)

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by matthewp** [comment]
We still need an example. Even if it doesn't break in Stackblitz we use these examples to pull down locally and eventually turn into a test.

**Comment by ohrrkan** [comment]
> We still need an example. Even if it doesn't break in Stackblitz we use these examples to pull down locally and eventually turn into a test.

linked

**Comment by mingjunlu** [comment]
Hi @ohrrkan,

I downloaded the repo you provided and tested locally with Safari 17.6. The cookie was successfully deleted after visiting http://localhost:4321/remove.

Here are the steps I take:
1. Visit http://localhost:4321/add
2. Notice that the cookie is set
3. Visit http://localhost:4321/remove
4. Notice that the cookie is removed

Is there anything that I might have missed in order to reproduce the situation?

**Comment by ascorbic** [comment]
I've also tried and failed to reproduce it with your code using the same steps as @mingjunlu. Can you provide step-by-step instructions to reproduce this.

**Comment by mingjunlu** [comment]
Sorry for not reading the description carefully. I actually reproduced it without noticing 😅

So the problem is that, even though the cookie is gone in the storage tab of Safari's developer tools, it is somehow still accessible on the client side (by accessing `document.cookie`).

Compared with Chrome and Firefox, the removed cookie is not accessible on the client side.

I did a little bit of searching and found a similar issue:
https://bugs.webkit.org/show_bug.cgi?id=247482

Seems like a Safari bug to me 🤔
